test/pipe.cc

```diff
@@ -0,0 +1,66 @@
+#include "eventuals/pipe.h"
+
+#include <thread>
+#include <vector>
+
+#include "eventuals/collect.h"
+#include "eventuals/terminal.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using eventuals::Collect;
+using eventuals::Pipe;
+
+TEST(Pipe, UniqueValue) {
+  Pipe<int> pipe;
+
+  *pipe.Write(1);
+  *pipe.Close();
+
+  auto e = [&pipe]() {
+    return pipe.Read()
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_EQ(std::vector<int>{1}, *e());
+}
+
+
+TEST(Pipe, Values) {
+  Pipe<int> pipe;
+
+  std::thread t([&pipe]() {
+    for (size_t i = 1; i <= 5; ++i) {
+      *pipe.Write(i);
+    }
+    *pipe.Close();
+  });
+
+  auto e = [&pipe]() {
+    return pipe.Read()
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_EQ(
+      (std::vector<int>{1, 2, 3, 4, 5}),
+      *e());
```

Comment:
```suggestion
  EXPECT_THAT(*e(), ElementsAre(1, 2, 3, 4, 5));
```
See for example how it's done in [test/range.cc](https://github.com/3rdparty/eventuals/blob/main/test/range.cc).